utils/ccdatofhir/mappings_allergy_intolerance.bal

```diff
@@ -228,3 +259,107 @@ isolated function mapCcdaValueToFhirAllergyIntoleranceType(xml valueElement) ret
         }
     }
 }
+
+isolated function mapCcdaSeverityToFhirSeverity(xml valueElement) returns uscore501:USCoreAllergyIntoleranceReactionSeverity? {
+    string|error? codeVal = valueElement.code;
+    if codeVal !is string {
+        log:printDebug("severity code value not available", codeVal);
+        return ();
+    }
+    match codeVal {
+        "255604002" => { // Mild
+            return uscore501:CODE_SEVERITY_MILD;
+        }
+        "6736007" => { // Moderate
+            return uscore501:CODE_SEVERITY_MODERATE;
+        }
+        "24484000" => { // Severe
+            return uscore501:CODE_SEVERITY_SEVERE;
+        }
+        _ => {
+            log:printDebug("matching severity code value not available");
+            return ();
+        }
+    }
+}
+
+# Map C-CDA telecom to FHIR ContactPoint.
+#
+# + telecomElement - C-CDA telecom element
+# + return - Return FHIR ContactPoint
+public isolated function mapCcdaTelecomToFhirContactPoint(xml telecomElement) returns r4:ContactPoint? {
+    string|error? telecomUse = telecomElement.use;
+    string|error? telecomValue = telecomElement.value;
+
+    string? systemVal = ();
+    string? valueVal = ();
+    if telecomValue is string {
+        string[] valTokens = re `:`.split(telecomValue);
+        if valTokens.length() == 1 {
+            systemVal = r4:other;
+            valueVal = telecomValue;
+        } else {
+            systemVal = valTokens[0];
+            valueVal = valTokens[1];
+            
+            match (systemVal) {
+                "tel" => {
+                    systemVal = r4:phone;
+                }
+                "mailto" => {
+                    systemVal = r4:email;
+                }
+                "fax" => {
+                    systemVal = r4:fax;
+                }
+                "http" => {
+                    systemVal = r4:url;
+                }
+                "x-text-fax" => {
+                    systemVal = r4:sms;
+                }
+                _ => {
+                    systemVal = r4:other;
+                }
+            }
+        }
+    } else {
+        log:printDebug("Telecom value not available", telecomValue);
+    }
+
+    r4:ContactPointUse? useVal = ();
+    if telecomUse is string {
+        match telecomUse {
+            "HP" => {
+                useVal = "home";
+            }
+            "WP" => {
+                useVal = "work";
+            }
+            "MC" => {
+                useVal = "mobile";
+            }
+            "OP" => {
+                useVal = "old";
+            }
+            "TP" => {
+                useVal = "temp";
+            }
+            "PG" => {
+                useVal = "work";
```

Comment:
Don't we need to have a default case here?

Reply:
It seems we cannot map to a default value as it may be a misinterpretation as mapping guide doesn't provide such guidance